### PR TITLE
Add configure support to enable RAM class persistence

### PR DESCRIPTION
**Pull request by ThanHenderson**
Adds support for setting the `J9VM_OPT_SNAPSHOTS` directive with the `--enable-snapshots` configure option. This enables the ongoing RAMClass persistence work.

Related: https://github.ibm.com/runtimes/openj9/issues/755
Signed-off-by: Nathan Henderson <nathan.henderson@ibm.com>

### CODE REVIEW

**Comment by ThanHenderson** [comment]
ping @babsingh

**Comment by keithc-ca** [comment]
Please correct the typo in the commit message and the summary here to spell "configure" correctly.

File: closed/OpenJ9.gmk [hunk 1]
```diff
@@ -391,6 +400,12 @@ else # OPENJ9_ENABLE_OPENJDK_METHODHANDLES
   CMAKE_ARGS += -DJ9VM_OPT_OPENJDK_METHODHANDLE=OFF
 endif # OPENJ9_ENABLE_OPENJDK_METHODHANDLES
 
+ifeq (true,$(OPENJ9_ENABLE_SNAPSHOTS))
+    CMAKE_ARGS += -DJ9VM_OPT_SNAPSHOTS=ON
+  else # OPENJ9_ENABLE_SNAPSHOTS
+    CMAKE_ARGS += -DJ9VM_OPT_SNAPSHOTS=OFF
+  endif # OPENJ9_ENABLE_SNAPSHOTS
+
```
Comment (babsingh) [inline closed/OpenJ9.gmk:408]: spacing is off
```suggestion
ifeq (true,$(OPENJ9_ENABLE_SNAPSHOTS))
  CMAKE_ARGS += -DJ9VM_OPT_SNAPSHOTS=ON
else # OPENJ9_ENABLE_SNAPSHOTS
  CMAKE_ARGS += -DJ9VM_OPT_SNAPSHOTS=OFF
endif # OPENJ9_ENABLE_SNAPSHOTS

```

Comment (ThanHenderson) [reply closed/OpenJ9.gmk]: Fixed.

**Comment by babsingh** [comment]
Jenkins compile aix,osx,win,zlinux jdk8

**Review by babsingh — APPROVED** [review]

File: jdk/make/closed/autoconf/custom-hook.m4 [hunk 2]
```diff
@@ -375,6 +376,25 @@ AC_DEFUN([OPENJ9_CONFIGURE_OPENJDK_METHODHANDLES],
...
+[
+  AC_MSG_CHECKING([for snapshot support])
+  AC_ARG_ENABLE([snapshots], [AS_HELP_STRING([--enable-snapshots],
+      [enable RAM persistence snapshots @<:@disabled@:>@])])
+  OPENJ9_ENABLE_SNAPSHOTS=false
+  if test "x$enable_snapshots" = xyes ; then
+    AC_MSG_RESULT([yes (explicitly enabled)])
+    OPENJ9_ENABLE_SNAPSHOTS=true
+  elif test "x$enable_snapshots" = xno ; then
+    AC_MSG_RESULT([no (explicit)])
```
Comment (keithc-ca) [inline jdk/make/closed/autoconf/custom-hook.m4:389]: Why does this not say `"no (explicitly disabled)"` like other options?

File: jdk/make/closed/autoconf/custom-spec.gmk.in [hunk 3]
```diff
@@ -61,6 +61,7 @@ OPENJ9_ENABLE_DEMOS                 := @OPENJ9_ENABLE_DEMOS@
 OPENJ9_ENABLE_JFR                   := @OPENJ9_ENABLE_JFR@
 OPENJ9_ENABLE_JITSERVER             := @OPENJ9_ENABLE_JITSERVER@
 OPENJ9_ENABLE_OPENJDK_METHODHANDLES := @OPENJ9_ENABLE_OPENJDK_METHODHANDLES@
+OPENJ9_ENABLE_SNAPSHOTS		    := @OPENJ9_ENABLE_SNAPSHOTS@
```
Comment (keithc-ca) [inline jdk/make/closed/autoconf/custom-spec.gmk.in:64]: Tabs should only be used for indentation (not after any non-whitespace character).

File: jdk/make/closed/autoconf/custom-hook.m4 (hunk 2 above)
Comment (ThanHenderson) [reply jdk/make/closed/autoconf/custom-hook.m4:389]: You're right. This is a mistake. I will open new PR to address any concerns you have here.

**Comment by ThanHenderson** [comment]
> Neither the UMA flag opt_snapshots nor the cmake build configuration flag J9VM_OPT_SNAPSHOTS exist: How can this have any effect?

`J9VM_OPT_SNAPSHOTS` is added in the accompanying [OpenJ9 patch](https://github.com/eclipse-openj9/openj9/pull/20387). (we don't need `opt_snapshots`; its presence here is a mistake)

> Why was it not first proposed for the head stream and then back-ported to earlier versions?

It is part of a revival of an older project that was/is only currently supporting JDK8.

**Comment by keithc-ca** [comment]
Reference to https://github.com/eclipse-openj9/openj9/pull/20387 and an explanation why it only applies to jdk8 would have been welcome pieces of information to include in the description here (although I have trouble imagining why it needs to be restricted to jdk8).

**Comment by ThanHenderson** [comment]
Apologies, I should have linked it.

It won't be restricted to JDK8, but it is only currently built/tested on JDK8. We are planning to port to new LTS JDK versions once we assess the requirements/feature support that is needed to accommodate them.

**Comment by keithc-ca** [comment]
> we don't need `opt_snapshots`; its presence here is a mistake

The code to adjust `opt_snapshots` is appropriate for UMA builds.

**Comment by ThanHenderson** [comment]
> The code to adjust opt_snapshots is appropriate for UMA builds.

I mean that it isn't used at all, nor specified, in the OpenJ9 patch, and we can add it later if we need.

**Pull request merged**